Xlib/ext/res.py

```diff
@@ -0,0 +1,285 @@
+            opcode=self.display.get_extension_major(extname),
+            client_major=client_major,
+            client_minor=client_minor)
+
+
+Client = rq.Struct(
+        rq.Card32("resource_base"),
+        rq.Card32("resource_mask"))
+
+
+class QueryClients(rq.ReplyRequest):
+    _request = rq.Struct(
+            rq.Card8("opcode"),
+            rq.Opcode(ResQueryClients),
+            rq.RequestLength())
+    _reply = rq.Struct(
+            rq.ReplyCode(),
+            rq.Pad(1),
+            rq.Card16("sequence_number"),
+            rq.ReplyLength(),
+            rq.LengthOf("clients", 4),
+            rq.Pad(20),
+            rq.List("clients", Client))
+
+
+def query_clients(self):
+    """Request the list of all currently connected clients."""
+    return QueryClients(
+            display=self.display,
+            opcode=self.display.get_extension_major(extname))
+
+
+Type = rq.Struct(
+        rq.Card32("resource_type"),
+        rq.Card32("count"))
+
+
+class QueryClientResources(rq.ReplyRequest):
+    _request = rq.Struct(
+            rq.Card8("opcode"),
+            rq.Opcode(ResQueryClientResources),
+            rq.RequestLength(),
+            rq.Card32("client"))
+    _reply = rq.Struct(
+            rq.ReplyCode(),
+            rq.Pad(1),
+            rq.Card16("sequence_number"),
+            rq.ReplyLength(),
+            rq.LengthOf("types", 4),
+            rq.Pad(20),
+            rq.List("types", Type))
+
+
+def query_client_resources(self, client):
+    """Request the number of resources owned by a client.
+
+    The server will return the counts of each type of resource.
+    """
+    return QueryClientResources(
+            display=self.display,
+            opcode=self.display.get_extension_major(extname),
+            client=client)
+
+
+class QueryClientPixmapBytes(rq.ReplyRequest):
+    _request = rq.Struct(
+            rq.Card8("opcode"),
+            rq.Opcode(ResQueryClientPixmapBytes),
+            rq.RequestLength(),
+            rq.Card32("client"))
+    _reply = rq.Struct(
+            rq.ReplyCode(),
+            rq.Pad(1),
+            rq.Card16("sequence_number"),
+            rq.ReplyLength(),
+            rq.Card32("bytes"),
+            rq.Card32("bytes_overflow"),
+            rq.Pad(16))
+
+
+def query_client_pixmap_bytes(self, client):
+    """Query the pixmap usage of some client.
+
+    The returned number is a sum of memory usage of each pixmap that can be
+    attributed to the given client.
+    """
+    return QueryClientPixmapBytes(
+            display=self.display,
+            opcode=self.display.get_extension_major(extname),
+            client=client)
+
+
+class SizeOf(rq.LengthOf):
+    """A SizeOf stores the size in bytes of some other Field whose size
+    may vary, e.g. List
+    """
+    def __init__(self, name, size, item_size):
+        rq.LengthOf.__init__(self, name, size)
+        self.item_size = item_size
+
+    def parse_value(self, length, _):
```

Comment:
Param `_` could be named `display` as for all other definitions. No matter it is not in use.

Reply:
Sure, changed to `display`.